src/NServiceBus.Core/Licensing/LicenseManager.cs

```diff
@@ -63,40 +60,58 @@ static void LogFindResults(ActiveLicenseFindResult result)
             }
         }
 
-        void PromptUserForLicenseIfTrialHasExpired()
+        void OpenTrialExtensionPage()
         {
+            string url;
+
+            if (result.License.IsExtendedTrial)
+            {
+                url = "https://particular.net/extend-your-trial-45";
+            }
+            else
+            {
+                url = "https://particular.net/extend-nservicebus-trial";
+            }
+
             if (!(Debugger.IsAttached && Environment.UserInteractive))
             {
-                //We only prompt user if user is in debugging mode and we are running in interactive mode
+                Logger.WarnFormat("To extend your trial license, visit: {0}", url);
+
                 return;
             }
 
-#if NET452
-            bool createdNew;
-            using (new Mutex(true, $"NServiceBus-{GitFlowVersion.MajorMinor}", out createdNew))
+            using (var mutex = new Mutex(true, @"Global\NServiceBusLicensing", out var acquired))
             {
-                if (!createdNew)
+                if (acquired)
                 {
-                    //Dialog already displaying for this software version by another process, so we just use the already assigned license.
-                    return;
-                }
+                    try
+                    {
+                        Logger.WarnFormat("Opening browser to: {0}", url);
 
-                if (license == null || LicenseExpirationChecker.HasLicenseExpired(license))
-                {
-                    var licenseProvidedByUser = LicenseExpiredFormDisplayer.PromptUserForLicense(license);
+                        var opened = Browser.Open(url);
+
+                        if (!opened)
+                        {
+                            Logger.WarnFormat("Unable to open browser. To extend your trial license, visit: {0}", url);
+                        }
 
-                    if (licenseProvidedByUser != null)
+                        Task.Delay(TimeSpan.FromSeconds(5)).GetAwaiter().GetResult();
+                    }
+                    finally
                     {
-                        license = licenseProvidedByUser;
+                        mutex.ReleaseMutex();
```

Comment:
As mentioned during the sync, there is a 5 second delay inside the mutex to prevent there being a race where each endpoint can acquire the mutex and open a browser.